tests/python/test_offset.py

```diff
@@ -49,3 +49,29 @@ def test():
     for i in range(16, 48):
         for j in range(-16, 16):
             assert a[i, j] == i + j * 10
+
+
+@ti.all_archs
+def test_offset_for_matrix():
+    a = ti.Matrix(dt=ti.i32, shape=(32, 16, 8), offset=(-8, -16, -32))
+    b = ti.Matrix(dt=ti.i32, shape=(32, 16, 8), offset=None)
+
+
+@ti.all_archs
+def test_offset_for_var():
+    a = ti.var(dt=ti.i32, shape=16, offset=-48)
+    b = ti.var(dt=ti.i32, shape=(16, ), offset=(16, ))
+    c = ti.var(dt=ti.i32, shape=(16, 64), offset=(-16, -64))
+    d = ti.var(dt=ti.i32, shape=(16, 64), offset=None)
+
+
+@ti.must_throw(AssertionError)
```

Comment:
Should it be `ValueError`?

Reply:
Yes, it should be indeed. Just now I convert all ValueError to AssertError, would it be ok?